# Ce travail risque encore de me donner des sueurs froides! 

In [1]:
%matplotlib inline
import pandas as pd
import numpy as np
import matplotlib as mp

In [2]:
pd.set_option("display.float_format", lambda x : "%.2f" % x)

In [18]:
ati = pd.read_csv("ati.csv")

In [19]:
ati

year  month request_number  \
0      2016      3   A-2015-00070   
1      2015      4   A-2015-00001   
2      2015      4   A-2014-00219   
3      2015      4   A-2014-00218   
4      2015      4   A-2014-00217   
5      2015      4   A-2014-00216   
6      2015      4   A-2014-00214   
7      2015      4   A-2014-00210   
8      2015      4   A-2014-00213   
9      2015      6   A-2015-00012   
10     2015      4   A-2014-00212   
11     2015      4   A-2014-00200   
12     2015      4   A-2014-00140   
13     2015      5   A-2015-00003   
14     2015      4   A-2014-00112   
15     2015      5   A-2014-00211   
16     2015      4   A-2014-00098   
17     2015      4   A-2014-00070   
18     2015      4   A-2015-00009   
19     2015      5   A-2015-00011   
20     2015      5   A-2015-00010   
21     2015      5   A-2015-00006   
22     2015      5   A-2014-00209   
23     2015      5   A-2014-00145   
24     2015      5   A-2014-00137   
25     2015      5   A-2014-00134   
26     2015      5   A-2014-00120   
27     2015      7   A-2015-00016   
28     2015      6   A-2015-00029   
29     2015      6   A-2015-00028   
...     ...    ...            ...   
28731  2016      1  AV-2015-00010   
28732  2016      1  AV-2015-00011   
28733  2016      2  AV-2015-00012   
28734  2015      5  AV-2014-00007   
28735  2015      7  AV-2015-00002   
28736  2015      9  AV-2015-00003   
28737  2015     10  AV-2015-00004   
28738  2016      1  AV-2015-00009   
28739  2016      6  AV-2016-00001   
28740  2016      4        2015-01   
28741  2016      4        2016-01   
28742  2017      1            NaN   
28743  2016     10        2016-06   
28744  2016     10        2016-07   
28745  2016     11        2016-20   
28746  2017      2        2016-22   
28747  2015      5   A-2014-00012   
28748  2016     10   A-2016-00003   
28749  2016     10   A-2016-00006   
28750  2016     10   A-2016-00007   
28751  2015      6   A-2015-00001   
28752  2015      3   A-2014-00011   
28753  2015      9   A-2015-00002   
28754  2015      3   A-2014-00009   
28755  2015      3   A-2014-00013   
28756  2016     11   A-2016-00004   
28757  2016      8   A-2016-00002   
28758  2016      6   A-2016-00001   
28759  2017      2   A-2016-00011   
28760  2017      2   A-2016-00013   

                                              summary_en  \
0      Provide a copy of the Transitional Briefing Bo...   
1      Briefing notes or memoranda submitted January ...   
2      Please provide from January 1, 2013 to present...   
3      All Temporary Help Services (THS) Supply Arran...   
4      All Task Based Professional Services (TSPS) an...   
5      Please provide all ProServices contracts issue...   
6      A list of all briefing material provided to th...   
7      Briefing material to the deputy minister, DMM ...   
8      A list of all briefing material provided to th...   
9      Seeking a copy of the Environmental Assessment...   
10     Briefing note to the minister, M1N 208144 - De...   
11     Would like the following: 2013-08-07 - UPDATE ...   
12     Records including but not limited to question ...   
13     Departmental guidelines, practices, frameworks...   
14     All documents related to and from before, duri...   
15     Bid award summary, including bidding companies...   
16     Please provide the briefing materials identifi...   
17     Wish to obtain all documents and information r...   
18     I would like the list of projects approved und...   
19     A copy of the TBIPS Tier 1 SA Contract (# 7200...   
20     I am looking for a genomics proposal (latest d...   
21     All Temporary Help Services (THS) Supply Arran...   
22     Please provide electronic copies of the briefi...   
23     All documents, including but not limited to em...   
24     Please provide an electronic copies of the fol...   
25     I would like access to documents about any pro...   
26     Copies of all correspondence, emails, fax's an...   
27     I am requesting a copy of

Ceci est pour savoir la longueur de mon fichier

In [20]:
ati.shape

(28761, 9)

Ceci est pour savoir les libellés des colonnes où chercher de l'information

In [21]:
ati.columns

Index(['year', 'month', 'request_number', 'summary_en', 'summary_fr',
       'disposition', 'pages', 'owner_org', 'owner_org_title'],
      dtype='object')

Pour savoir un peu le type de données qui se trouvent à l'intérieur du fichier

In [22]:
ati.dtypes

year                int64
month               int64
request_number     object
summary_en         object
summary_fr         object
disposition        object
pages               int64
owner_org          object
owner_org_title    object
dtype: object

On se rend compte que, sur les 28761 articles du fichier, certaines données sont manquantes (exemple: request_numbers), et qu'il y a plus de sommaires en anglais qu'en français (ah, le bilinguisme!)

In [26]:
ati.count()

year               28761
month              28761
request_number     28748
summary_en         28741
summary_fr         28667
disposition        28740
pages              28761
owner_org          28761
owner_org_title    28761
dtype: int64

Pour savoir les données s'étendent entre quelles années

In [27]:
ati.year.min()

2015

In [28]:
ati.year.max()

2017

Pour savoir si les données qu'on a sont partielles ou s'il y en de de cachées (ah, le gouvernement!)

In [29]:
ati.disposition

0        Disclosed in part / Communication partielle
1         No records exist / Aucun document existant
2         No records exist / Aucun document existant
3            Disclosed in part / Divulgués en partie
4         No records exist / Aucun document existant
5            Disclosed in part / Divulgués en partie
6            Disclosed in part / Divulgués en partie
7                         All excluded / Tous exclus
8            Disclosed in part / Divulgués en partie
9        Disclosed in part / Communication partielle
10           Disclosed in part / Divulgués en partie
11           Disclosed in part / Divulgués en partie
12           Disclosed in part / Divulgués en partie
13              All disclosed / Communication totale
14           Disclosed in part / Divulgués en partie
15       Disclosed in part / Communication partielle
16           Disclosed in part / Divulgués en partie
17           Disclosed in part / Divulgués en partie
18         Request transferred / Demande trans

Pour savoir qui sont les ministère les plus fréquemment interrogés.

In [33]:
ati.owner_org.value_counts()

dnd-mdn         2605
dfatd-maecd     1650
esdc-edsc       1541
cic             1323
hc-sc           1164
rcmp-grc        1121
pwgsc-tpsgc     1116
cra-arc         1046
tc              1026
fin             1011
pco-bcp          997
aandc-aadnc      911
lac-bac          897
ic               893
cbsa-asfc        872
nrcan-rncan      783
jus              737
dfo-mpo          713
pch              637
ps-sp            627
ec               616
tbs-sct          578
cfia-acia        498
csc-scc          420
vac-acc          372
ssc-spc          346
csis-scrs        335
infc             283
cnsc-ccsn        254
aafc-aac         245
                ... 
wd-deo            14
jccbi-pjcci       12
ocl-cal           12
pmprb-cepmb       11
cannor            11
vrab-tacra        10
bdc                9
pspib-oirpsp       8
fcac-acfc          7
cpc-cpp            7
petf-fpet          7
wdba-apwd          7
prpa-appr          5
cdic-sadc          4
erc-cee            4
fcc-fac            3
glpa-apgl    

Pour un peu voir le taux d'information divulguée en totalement, en partie ou pas du tout à la suite des demandes d'accès à l'information.

In [32]:
ati.disposition.value_counts()

Disclosed in part / Communication partielle                                                                                                                                                                                                                                                                                                    8653
All disclosed / Communication totale                                                                                                                                                                                                                                                                                                           2907
Disclosed in part                                                                                                                                                                                                                                                                                                               

Pour savoir le pourcentage de données divulguées, tous les organismes confondus

In [41]:
ati.disposition.value_counts() / len(ati.disposition)*100

Disclosed in part / Communication partielle                                                                                                                                                                                                                                                                                                   30.09
All disclosed / Communication totale                                                                                                                                                                                                                                                                                                          10.11
Disclosed in part                                                                                                                                                                                                                                                                                                               